lib/portage/package/ebuild/doebuild.py

```diff
@@ -317,24 +305,15 @@ def _doebuild_path(settings, eapi=None):
 
     for x in portage_bin_path:
         path.append(os.path.join(x, "ebuild-helpers"))
-    path.extend(prerootpath)
-
-    for prefix in prefixes:
-        prefix = prefix if prefix else "/"
-        for x in (
-            "usr/local/sbin",
-            "usr/local/bin",
-            "usr/sbin",
-            "usr/bin",
-            "sbin",
-            "bin",
-        ):
-            # Respect order defined in ROOTPATH
-            x_abs = os.path.join(prefix, x)
-            if x_abs not in rootpath_set:
-                path.append(x_abs)
 
-    path.extend(rootpath)
+    # If PATH is set in env.d, ignore PATH from the calling environment.
+    # This allows packages to update our PATH as they get installed.
+    if "PATH" in settings.configdict["env.d"]:
+        settings.configdict["env"].pop("PATH", None)
+
+    if "PATH" in settings:
+        path.extend(settings["PATH"].split(":"))
```

Comment:
This looks like it could lead to feedback, causing unbounded growth of the PATH setting with duplicate values.

Reply:
It does seem that way, especially if env.d is empty and we skip line 312.

However, I wasn't able to cause PATH to be duplicated, even after commenting out line 311 and 312. I assume that the _doebuild_path function simply isn't called more than once with the same config instance.

I could add a failsafe If you think it makes sense. Maybe something like `settings["__doebuild_path"] = "done"`, and make `_doebuild_path` a noop if it finds that setting.

Reply:
I changed the code to avoid adding duplicate entries. That's probably less error-prone.

Reply:
I think I'd feel better if we had one, to be robust to future changes. Or a test to try check that duplicates get filtered if it doesn't already?

Reply:
> However, I wasn't able to cause PATH to be duplicated, even after commenting out line 311 and 312.

The PATH was actually getting duplicated during the "merge" phase, so I didn't see it in my ebuild-based tests. The revised code with duplicate elimination should fix that.